### PR TITLE
Add some physics into about lower-hybrid frequency

### LINKED ISSUE #1172: Describe plasma physics behind upper and lower hybrid frequencies
The docstrings for `upper_hybrid_frequency` and `lower_hybrid_frequency` in `plasmapy.formulary` should be expanded to include a description of the physics related to these frequencies, including how they fit in with plasma waves.

### PR DESCRIPTION
**Pull request by alfkoehn**
Triggered by the hack-week, I added a little bit of physics information about the lower-hybrid resonance.

Together with #1180, this should close #1172 

- [x] I have added a changelog entry for this pull request.

- [ ] If adding new functionality, I have added tests and
      docstrings.

- [ ] I have fixed any newly failing tests.

### CODE REVIEW

File: plasmapy/formulary/parameters.py [hunk 1]
```diff
@@ -1863,6 +1863,14 @@ def lower_hybrid_frequency(B: u.T, n_i: u.m ** -3, ion: Particle) -> u.rad / u.s
     :math:`ω_{ce}` is the electron gyrofrequency, and
     :math:`ω_{pi}` is the ion plasma frequency.
 
+    The lower hybrid frequency consitutes a resonance for electromagnetic
+    waves in magnetized plasmas, namely for the X-mode. There are waves
```
Comment (namurphy) [inline plasmapy/formulary/parameters.py:1867]:
```suggestion
    waves in magnetized plasmas, namely for the X-mode. These are waves
```

**Review by namurphy — COMMENTED** [review]
Thank you again!  One suggested wording change, and otherwise looks good.

File: plasmapy/formulary/parameters.py (hunk 1 above)
Comment (alfkoehn) [reply plasmapy/formulary/parameters.py:1867]: sorry about that typo, will fix it asap

**Review by namurphy — APPROVED** [review]
Great, thank you once again!

**Pull request merged**